src/Tuple.js

```diff
@@ -0,0 +1,74 @@
+function Tuple(x, y) {
+    switch (arguments.length) {
+        case 0:
+            throw new TypeError('no arguments to Tuple');
+        case 1:
+            return function(y) {
+                return new _Tuple(x, y)
+            };
+        default:
+            return new _Tuple(x,y)
+    }
+};
+
+function _Tuple(x, y) {
+    this[0] = x;
+    this[1] = y;
+    this.length = 2;
+};
+
+function ensureConcat() {
+    var xs = [].slice.call(arguments)
+    xs.map(function(x){
+      if(typeof x.concat != "function") {
+          var display = x.show ? x.show() : x;
+          throw new TypeError(display + " must be a semigroup to perform this operation");
+      };
+    })
+}
+
+Tuple.of = function(x) {
+    return Tuple(x, x);
```

Comment:
It should be `(empty, x)` since `snd` is the functor target, but that requires `Monoid a => Tuple a b` which is only necessary for the applicative instance.